benchmarks/swe_bench/swe_bench.py

```diff
@@ -0,0 +1,374 @@
+        patch_path = os.path.join(path_to_logs, result, "patch.diff")
+
+        if os.path.exists(results_path) and os.path.exists(patch_path):
+            # Sometimes there is no result saved, at which point we ignore that entry
+            with open(results_path, "r") as f:
+                result_dict = json.load(f)
+                instance_id, raw_results = next(iter(result_dict.items()))
+
+                with open(patch_path, "r") as f:
+                    results_per_instance_id[instance_id] = {
+                        "resolved": raw_results["resolved"],
+                        "patch": f.read(),
+                    }
+
+    @scorer(metrics=[mean(), std()], name=baseline_name)
+    def _swebench_baseline_scorer() -> Scorer:
+        async def scorer(state: TaskState, target: Target) -> Score:
+            if state.sample_id in results_per_instance_id:
+                results = results_per_instance_id[state.sample_id]
+                return Score(
+                    value=results["resolved"],
+                    explanation=f"Model Patch:\n\n {results["patch"]}",
+                )
+            else:
+                return Score(
+                    value="N", explanation="No baseline found for this instance"
+                )
+
+        return scorer
+
+    return _swebench_baseline_scorer()
+
+
+def get_compose_file(environment_commit_id: Sample, instance_id: str) -> str:
+    if not os.path.exists(SAMPLE_TO_IMAGE_PATH):
+        raise ValueError(
+            f"No sample to image mapping found. Please run  to build the images"
+        )
+
+    sample_to_image = json.load(open(SAMPLE_TO_IMAGE_PATH))
+    if (
+        instance_id in sample_to_image
+        and environment_commit_id in sample_to_image[instance_id]
+    ):
+        environment_image_name = sample_to_image[instance_id][environment_commit_id][
+            "image_key"
+        ]
+    else:
+        raise ValueError(
+            f"No image found for instance_id {instance_id}. Please run 'build_images.py --dataset_location DATASET_LOCATION --split SPLIT' to build the images"
+        )
+
+    compose_file_path = f"{COMPOSE_FILE_DIR}/{environment_commit_id}.yaml"
+    if os.path.exists(compose_file_path):
+        return compose_file_path
+
+    images = DockerClient.from_env().images.list()
+    if environment_image_name not in [
+        image_name for image in images for image_name in image.tags
+    ]:
+        raise ValueError(
+            f"Image {environment_image_name} not found in docker images. Please run 'build_images.py --dataset_location DATASET_LOCATION --split SPLIT' to build the images"
+        )
+
+    # If the image is found, we can now create the compose file.
+    compose_file_path = COMPOSE_FILE_DIR / f"{environment_image_name}.yaml"
+    with compose_file_path.open(mode="w+") as f:
+        f.write(f"""services:
+  default:
+    image: {environment_image_name}
+    command: "sleep infinity"
+    working_dir: /testbed
```

Comment:
I did sleep infinite instead of the "tail -f" version, since I think that this responds better to interrupts so should make everythiing shut down faster... Although this hasn't actually happened (maybe I'm wrong?)